OneCommunityGlobal/HighestGoodNetworkApp
# Haoji create profile picture selection for admin

# Description
<img width="664" alt="image" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/99279909/84737f22-f684-46f7-abbd-0661ff8352b4">

## Related PRS (if any):
This frontend PR is related to the [#591](https://github.com/OneCommunityGlobal/HGNRest/pull/591) backend PR.
…

## Main changes explained:
- Update src/actions/userManagement.js and add a new column for owners to manage the profile pictures
- Create src/actions/userManagement.js for admin to select proper profile picture that matches the user
…

## How to test:
1. check into current branch
2. do `npm install` and `npm run start:local` to run this PR locally
3. See backend how to setup [#591](https://github.com/OneCommunityGlobal/HGNRest/pull/591), change the cronjob period 
4. log in as owner user
5. go to other link→ user management → Profile Image →…
6. Create a new user according to the name on [https://www.onecommunityglobal.org/team/](https://www.onecommunityglobal.org/team/). If you cannot see your name on the site, feel free to create a new user with the following cases. Then wait for the cronjob execution 
7. Case 1: If the user already has a profile picture, the cronjob will skip and not overwrite the original picture. (You can paste an image URL into the "Profile Image" and click "add". Wait and see if cronjob will overwrite the picture)
8. Case 2: If the user name is unique and there is no profile picture, the cronjob automatically assigns one
9. Case 3: If the user name matches multiple images on the sites. For example, there is a person named Yifan Wang and another named Yi Wang, cronjob may not be able to tell the belongings of pictures. Owners in that case, could select the proper one from all possible pictures.

## Screenshots or videos of changes:

Case1 & 2: 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/99279909/ba908164-f8a0-43e2-87eb-7d2b0d6f56d0

Case 3:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/99279909/61aae8c2-c16f-4357-a435-b0d3bda612a4

## Note:
URL is currently valid and available for the profile picture
